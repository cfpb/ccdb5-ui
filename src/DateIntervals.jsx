import './DateIntervals.less'
import { connect } from 'react-redux'
import { dateIntervals } from './constants';
import { dateIntervalToggled } from './actions/filter';
import React from 'react'


export class DateIntervals extends React.Component {
  _setDateInterval( page ) {
    this.props.toggleDateInterval( page );
  }

  _btnClassName( dateInterval ) {
    const btnClass = 'date-selector ';
    return dateInterval === this.props.dateInterval ?
      btnClass + 'selected' : btnClass;
  }

  render() {
    return (
      <section className="date-intervals">
        { dateIntervals.map( dateInterval =>
          <button onClick={ () => this._setDateInterval( dateInterval ) }
                  className={ this._btnClassName( dateInterval ) }
                  key={dateInterval}>
            {dateInterval}
          </button>
        ) }
      </section>
    )
  }
}

export const mapStateToProps = state => ( {
  dateInterval: state.query.dateInterval
} );

export const mapDispatchToProps = dispatch => ( {
  toggleDateInterval: interval => {
    dispatch( dateIntervalToggled( interval ) )
  }
} );

export default connect( mapStateToProps, mapDispatchToProps )( DateIntervals )
