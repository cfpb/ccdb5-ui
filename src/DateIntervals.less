@import (less) "base.less";

.date-intervals {
  width: 100%;
  float: left;
  .date-selector {
    background-color: @pacific-40;
    &.selected {
      background-color: @pacific-80;
    }
  }
}