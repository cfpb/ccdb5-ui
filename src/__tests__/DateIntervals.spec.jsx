import configureMockStore from 'redux-mock-store'
import { mapStateToProps, DateIntervals } from '../DateIntervals'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'


function setupSnapshot() {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    map: {}
  } )

  return renderer.create(
    <Provider store={ store }>
      <DateIntervals />
    </Provider>
  )
}

describe( 'component: DateIntervals', () => {
  describe( 'initial state', () => {
    it( 'renders without crashing', () => {
      const target = setupSnapshot()
      let tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )
  } )

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        view: {
          dateInterval: 'foo'
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( { dateInterval: 'foo' } )
    } )
  } )


} )
