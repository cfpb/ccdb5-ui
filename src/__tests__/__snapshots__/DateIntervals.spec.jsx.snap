// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: DateIntervals initial state renders without crashing 1`] = `
<section
  className="date-intervals"
>
  <button
    className="date-selector "
    onClick={[Function]}
  >
    3m
  </button>
  <button
    className="date-selector "
    onClick={[Function]}
  >
    6m
  </button>
  <button
    className="date-selector "
    onClick={[Function]}
  >
    1y
  </button>
  <button
    className="date-selector "
    onClick={[Function]}
  >
    3y
  </button>
  <button
    className="date-selector "
    onClick={[Function]}
  >
    All
  </button>
</section>
`;
